integration-tests/src/tests/client/features/delegate_action.rs

```diff
@@ -108,6 +110,31 @@ fn reject_valid_meta_tx_in_older_versions() {
     );
 }
 
+/// For NEAR-implicit accounts, returns the corresponding implicit public key.
+/// For Named accounts, generates a public key using the account ID as the seed.
+///
+/// In case of ETH-implicit account, the ETH-implicit address shall be derived from the public key.
+/// Thus, we cannot derive the public key from ETH-implicit address and the caller must
+/// explicitly provide the Secp256K1 `public_key` matching the sender address.
+fn derive_public_key_from_sender(sender: &AccountId, public_key: Option<PublicKey>) -> PublicKey {
```

Comment:
It seems strange to me that this function is receiving the public key in the case it cannot be derived. I think we should skip calling this function entirely if we already have the public key available.